packages/messaging/lib/index.d.ts

```diff
@@ -1012,6 +1088,12 @@ export namespace FirebaseMessagingTypes {
      * @param enabled A boolean value to enable or disable exporting of message delivery metrics to BigQuery.
      */
     setDeliveryMetricsExportToBigQuery(enabled: boolean): Promise<void>;
+    /**
+     * Checks if all required APIs exist in the browser.
+     *
+     * @web
+     */
+    isSupported(): Promise<boolean>;
```

Comment:
This appears a little complicated. On native, it is supported on iOS in general, and on Android *if* our Utils method says google play services exist, so there is a way to implement it.

firebase-js-sdk definitely implements it - but dynamically (!) - by defining it after the getMessagingInWindow / getMessagingInSw definitions like so --> https://github.com/firebase/firebase-js-sdk/blob/49ee786f2b022e65aef45693e1a8b546d889ec10/packages/messaging/src/api.ts#L53

it appears we need to have the symbol available to be firebase-js-sdk compatible

Reply:
What API corresponds to this for iOS? For FlutterFire, we just return `true`: https://github.com/firebase/flutterfire/blob/master/packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/method_channel/method_channel_messaging.dart#L148

As for the web, not sure what more can be done at this moment in time until web is an available platform?